c++/src/kj/compat/http.h

```diff
@@ -662,9 +659,20 @@ class HttpClient {
   // `url` and `headers` need only remain valid until `openWebSocket()` returns (they can be
   // stack-allocated).
 
-  virtual kj::Promise<kj::Own<kj::AsyncIoStream>> connect(kj::StringPtr host);
-  // Handles CONNECT requests. Only relevant for proxy clients. Default implementation throws
-  // UNIMPLEMENTED.
+  struct ConnectResponse {
+    uint statusCode;
+    kj::StringPtr statusText;
+    const HttpHeaders* headers;
+    kj::Own<kj::AsyncIoStream> io;
```

Comment:
If the server response with an error, is the connection still established? I'd expect not. Can the error response contain a regular HTTP response body? If so then I think we need a `OneOf` here to allow for that, like in `WebSocketResponse`.

Reply:
See below. I've change the `connect()` method here to return `kj::Promise<kj::OneOf<ConnectResponse, Response>>`. Returning `Response` is an indication that the tunnel has not been established and the response is still HTTP rules. Returning `ConnectResponse` indicates that we're proceeding with the tunnel.

Reply:
Can we be consistent with how `openWebSocket()` works?

Reply:
Yep. Will get that updated next week.